src/platformio.ts

```diff
@@ -85,21 +97,85 @@ export class PlatformIO {
                 let includes = idedataJson.includes;
                 this.addIncludePathToFile(includes);
             } else {
-                this.showErrorToAddIncludePath();
+                if (!silent) {
+                    this.showErrorToAddIncludePath();
+                }
+
                 return;
             }
-            vscode.window.showInformationMessage("Add Include Path to c_cpp_properties.json successfully.");
+
+            if (!silent) {
+                vscode.window.showInformationMessage("Add Include Path to c_cpp_properties.json successfully.");
+            }
         } catch (ex) {
-            this.showErrorToAddIncludePath();
+            if (!silent) {
+                this.showErrorToAddIncludePath();
+            }
         } finally {
-            AppInsightsClient.sendEvent("addIncludePath");
+            if (!silent) {
+                AppInsightsClient.sendEvent("addIncludePath");
+            }
         }
     }
 
     public onDidCloseTerminal(closedTerminal: vscode.Terminal): void {
         this._pioTerminal.onDidCloseTerminal(closedTerminal);
     }
 
+    private initializeAutoUpdateIncludes(): void {

```

Comment:
If we want to turn on it by default, we also need to check whether user has installed the C/C++ extension, otherwise the c_cpp_properties.json is useless for user.